### ISSUE TITLE
SELF_SIGNED_CERT_IN_CHAIN

### ISSUE THREAD

**Issue by grelawit** [npm/npm#16457]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [x] npm is producing an incorrect install.
  - [x] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?
It's the old trouble of the not supported self-signed certificate. NPM fails to install, after I created a project through the Angular-CLI. I found quite a lot of fixes online, yet none of them helped (upgrading node f.e. . I then used node 6.10.2 and npm 3.10.10. I set the «npm config set ca=""», etc.). 

I'm kind of running out of ideas... I found the tip to point to a certificate, but to be honest, that confused me a little bit...

#### How can the CLI team reproduce the problem?

As the other fixes seem to work for people, I am not really sure tbh... Trouble is, I can't quite decipher the log file... But words like "verbose", "silly", and "ERR!" don't sound too comforting :( .

Said log can be found in this gist: https://gist.github.com/grelawit/8cc94c800a73ae99a809f850edbdf85b

### supporting information:

 - `npm -v` prints: 3.10.10
 - `node -v` prints: 6.10.2
 - `npm config get registry` prints: https://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?: OSX 10.10.5 (Might that be the problem?).
 - Network issues:
   - Geographic location where npm was run: Central Europe
   - [ ] I use a proxy to connect to the npm registry.
   - [ ] I use a proxy to connect to the web.
   - [ ] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
   -All above: Probably not, it worked fine earlier. But then I updated node (and npm) via Homebrew. The revert through nvm. I work in a private environment, no company proxy or so.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [ ] I develop using Vagrant on OS X or Linux.
   - [ ] I develop / deploy using Docker.
   - [ ] I deploy to a PaaS (Triton, Heroku).

**Comment by legodude17** [comment]
Have you taken a look at http://blog.npmjs.org/post/78165272245/more-help-with-selfsignedcertinchain-and-npm? See if the steps outlined there work for you.

If it does NOT:

1. Are you behind a proxy?
2. How are you trying to use it?

**Comment by grelawit** [comment]
**TL;DR: It's all Kaspersky's fault!!**

Hi @legodude17, and thank you so much for your help. Sorry for the late reply, I have been on holiday the last couple of days. 

I had seen this post, tried it before, and it had not worked. Ran two recommended commands again today. Then upgraded Node, and as I wanted to check wether I am in fact behind a Proxy I remembered how Kasperky Internet Security had once blocked me from work. Disabled it, and voilà! There was npm doing it's magic...

I feel really silly now... But, maybe, somebody will come here some day with the same problem and AV software.

I'll now have to find out how to get node back to work without disabling Kaspersky completely.

Greetings!

**Comment by legodude17** [comment]
Glad I could help! @KenanY this can be closed
